### PR TITLE
add node plugin and make it run frontend tests

### PR DESCRIPTION
**Pull request by renanfranca**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -17,6 +19,31 @@ springBoot {
...
+  npmVersion.set("10.5.2")
+}
+
+val testTaskUsingNpm = tasks.register<NpmTask>("testNpm") {
+  dependsOn("test", "integrationTest")
+  dependsOn(tasks.npmInstall)
+  npmCommand.set(listOf("run", "test"))
+//  args.set(listOf("test"))
+  ignoreExitValue.set(false)
+  environment.set(mapOf("APP_VERSION" to "0.0.1-SNAPSHOT")) //gradle doesnt use project version like maven. Where should I get that?
```
Comment (renanfranca) [inline build.gradle.kts]: I need help here: 
```kts
 environment.set(mapOf("APP_VERSION" to "0.0.1-SNAPSHOT")) //gradle doesnt use project version like maven. Where should I get that?
 ```

Comment (murdos) [reply build.gradle.kts]: It's the version defined here: https://github.com/renanfranca/sample-frontend-gradle/blob/45f55a752bb2e01177f08a820d1098f0e313ac71/build.gradle.kts#L28C1-L28C8
You can maybe access it with `project.version`

File: build.gradle.kts [hunk 2]
```diff
@@ -17,6 +19,31 @@ springBoot {
...
+  workingDir.set(projectDir)
+  execOverrides {
+    standardOutput = System.out
+  }
+  inputs.dir("node_modules")
+  inputs.file("package.json")
```
Comment (murdos) [inline build.gradle.kts]: There are a bit more input files:
- package-lock.json
- src/test/javascript
- src/main/webapp
- tsconfig*.json
- vite.config.*
- vitest.config.*

See e.g. for jhlite: https://github.com/jhipster/jhipster-lite/blob/3266169af076475fb0b00a6e0330128371414407/pom.xml#L912-L948

File: build.gradle.kts (hunk 1 above)
Comment (renanfranca) [reply build.gradle.kts]: Thank you!

**Pull request closed without merging**